spec.md

```diff
@@ -325,19 +330,28 @@ on the definition of OPTIONAL.
   event format defines the relationship in
   [section 3.1](./json-format.md#31-handling-of-data).
 
-  When this attribute is omitted, `data` simply follows the event
-  format's encoding rules. For the JSON event format, the `data` value
-  can therefore be a JSON object, array, or value.
-
-  For the binary mode of some of the CloudEvents transport bindings, where the
-  `data` content is immediately mapped into the payload of the transport frame,
-  this field is directly mapped to the respective transport or application
+  For the binary mode of some of the CloudEvents protocol bindings, where the
+  `data` content is immediately mapped into the payload of the protocol frame,
+  this field is directly mapped to the respective protocol or application
   protocol's content-type metadata property. Normative rules for the binary mode
-  and the content-type metadata mapping can be found in the respective transport
+  and the content-type metadata mapping can be found in the respective protocol
   mapping specifications.
 
+  This attribute MAY be omitted if the event format has an implied
+  `datacontenttype`. For example, the JSON format implies "application/json". A
+  JSON-format event with no `datacontenttype` is exactly equivalent to one with
+  `datacontenttype="application/json"` and its `data` can be any JSON object,
+  array, or value.
+
+  Formats and protocols that do not have an implied `datacontenttype` MUST NOT
+  omit this attribute. Agents that translate messages between formats or
+  protocols MUST ensure that an implied `datacontenttype` is made explicit in
+  the new format. For example when translating a JSON event with no
+  `datacontenttype` to AMQP, the AMQP `content-type` message property MUST be
+  set to "application/json".
+
```

Comment:
+1

Reply:
I think it's consistent with what the spec already says, and it rules out loopy interpretations like my previous "it must be a CE type" nonsense.

It does mean that you cannot use an AMQP value-section message as an event without additional clarifications in the AMQP binding spec and/or proprietary transformations - but that can be solved in the AMQP binding spec, and that's the right place to solve it

Reply:
@clemensv thoughts?

Reply:
I think this is discussed more below & hopefully addressed by latest update.